core/trino-parser/src/main/java/io/trino/sql/tree/QualifiedName.java

```diff
@@ -59,9 +59,11 @@ public static QualifiedName of(Iterable<Identifier> originalParts)
     private QualifiedName(List<Identifier> originalParts)
     {
         this.originalParts = originalParts;
-        this.parts = originalParts.stream()
-                .map(QualifiedName::mapIdentifier)
-                .collect(toImmutableList());
+        ImmutableList.Builder partsBuilder = ImmutableList.builder();
```

Comment:
Add a comment why it is implemented this way, otherwise someone surely soon will revert your change.